bin/spice/pkg/cli/cmd/login.go

```diff
@@ -138,10 +139,22 @@ spice login dremio --username <username> --password <password>
 			os.Exit(1)
 		}
 
+		endpoint, err := cmd.Flags().GetString(endpointFlag)
+		if err != nil {
+			cmd.Println(err.Error())
+			os.Exit(1)
+		}
+
+		if endpoint == "" {
+			cmd.Println("No endpoint provided, use --endpoint or -p to provide an endpoint")
+			os.Exit(1)
+		}
+
 		mergeAuthConfig(cmd, api.AUTH_TYPE_DREMIO, &api.Auth{
 			Params: map[string]string{
 				api.AUTH_PARAM_USERNAME: username,
 				api.AUTH_PARAM_PASSWORD: password,
+				api.AUTH_PARAM_ENDPOINT: endpoint,
```

Comment:
I think this should be a property of the dataset instead of a property of the auth, so `spice dataset configure` would ask for an endpoint and then write it to the dataset params. If auth determines the dataset location and not the dataset, then sharing spicepods becomes difficult because users will have to ensure that their auth configurations match.